docs/moz-addon-manager.md

```diff
@@ -0,0 +1,173 @@
+# Develop features for mozAddonManager
+
+The AMO and Discovery Pane apps use the
+[`mozAddonManager`](https://bugzilla.mozilla.org/show_bug.cgi?id=1310752)
+web API to achieve a more seamless add-on installation user experience.
+However, this API is only available to a limited list of domains.
+If you go to https://addons.mozilla.org (production) in Firefox then `mozAddonManager`
+is available on the page. If you go to a development site then it's not.
+
+## Turning on mozAddonManager for development
```

Comment:
> Also related is the xpinstall.signatures.dev-root pref required to allow install signed add-ons from -dev or stage

Oh, fun times! That will be an exercise for a future date. I am assuming the patch to make a custom Firefox allow add-on installations from localhost will not be easy.